pil-analyzer/src/pil_analyzer.rs

```diff
@@ -71,6 +74,12 @@ struct PILAnalyzer {
     symbol_counters: Option<Counters>,
     /// Symbols from the core that were added automatically but will not be printed.
     auto_added_symbols: HashSet<String>,
+    /// Implementations found, organized according to their associated trait name.
+    implementations: HashMap<String, Vec<TraitImplementation<Expression>>>,
+    /// A map between the name and type_Args of the caller reference
+    /// and the expression to be called.
+    /// Empty until resolve_trait_impls() is called.
+    solved_impls: HashMap<String, HashMap<Vec<Type>, Arc<Expression>>>,
```

Comment:
Could also be a local variable inside `analyze()` that is then passed to the `condense` function.